### ISSUE TITLE
SGD decisions

### ISSUE THREAD

**Issue by StephieX** [Microsoft/CNTK#1347]
I'm trying to figure out how BrainScript makes decisions concerning SGD iterations.
If I do this with 100 epochs 
    SGD = {
        epochSize = 0                   # =0 means size of the training set
        learningRatesPerSample = .1   # gradient contribution from each sample
        maxEpochs = 100
    }
I get these errors:

Final Results: Minibatch[1-1]: errs = 7.143% * 14; ce = 0.12205110 * 14; perplexity = 1.12981184

However increasing maxEpochs to 500 gives these errors:

Final Results: Minibatch[1-1]: errs = 78.571% * 14; ce = 1.88566235 * 14; perplexity = 6.59071837

Am I understanding the error printout correctly?  Both the local error, ce, and the aggregated error cerrs seem to be increasing with more epochs. 

It seems that the overall SGD algorithm shouldn't increase errors for subsequent iterations over the entire dataset.  I'm trying to figure out what the error reports means as much as anything.  For instance what does perplexity mean?

Thanks.

**Comment by cha-zhang** [comment]
It's likely your learning rate is to high (you are using learning rate per sample of 0.1, which is very high). In that case the SGD may be diverging.

**Comment by ebarsoumMS** [comment]
Try changing it to "learningRatesPerMB".

**Comment by mfuntowicz** [comment]
Hi @StephieX ,

Perplexity is just another metrics measuring how your model fits to your data.
The lower it is, the better.

 More information [here](https://en.wikipedia.org/wiki/Perplexity)

Morgan

**Comment by StephieX** [comment]
Thanks everybody for your help.  All your suggestions and comments were very helpful, especially for working with more realistic dataset sizes.  I was testing with around 15 samples and my actual dataset is over 200,000 samples.  Thanks.

**Comment by tangyuq** [comment]
Close this for now. Feel free to reopen it if you have further comments.
